qa/rpc-tests/wallet_sapling.py

```diff
@@ -201,5 +202,68 @@ def run_test(self):
         except JSONRPCException as e:
             assert_equal("Cannot send to both Sprout and Sapling addresses using z_sendmany", e.error['message'])
 
+        # Check z_sendmanywithchangetosender
+        # send from node 0 taddr to node 2 single taddr
+        senderaddr = self.nodes[0].getnewaddress()
+        self.nodes[0].generate(10)
+
+        mytxid = self.nodes[0].sendtoaddress(senderaddr, Decimal('100.0'))
+        senderExpectedBalance = Decimal('100.0')
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_all()
+        assert_equal(self.nodes[0].z_getbalance(senderaddr), senderExpectedBalance)
+
+        receiveraddr = self.nodes[2].getnewaddress()
+        receiveraddr_oldbalance = self.nodes[2].z_getbalance(receiveraddr)
+        fee         = Decimal('0.01')
+        self.sync_all()
+        minconf     = 1
+        receiverExpectedBalance = receiveraddr_oldbalance + Decimal('5.0')
+        recipients  = [ {"address": receiveraddr, "amount": Decimal('5.0')} ]
+
+        myopid = self.nodes[0].z_sendmanywithchangetosender(senderaddr, recipients, minconf, fee)
+        assert(myopid)
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_all()
+        assert_equal(self.nodes[2].z_getbalance(receiveraddr), receiverExpectedBalance)
+        calculatedSenderBalance = senderExpectedBalance - Decimal('5.0') - fee
+        assert_equal(self.nodes[0].z_getbalance(senderaddr), calculatedSenderBalance)
+
+        # send from node 0 taddr to node 2 multiple taddr
+        receiveraddr = self.nodes[2].getnewaddress()
+        receiveraddr_oldbalance = self.nodes[2].z_getbalance(receiveraddr)
+        receiveraddr2 = self.nodes[2].getnewaddress()
+        receiveraddr2_oldbalance = self.nodes[2].z_getbalance(receiveraddr2)
+
+        receiverExpectedBalance = receiveraddr_oldbalance + Decimal('5.0')
+        receiver2ExpectedBalance = receiveraddr2_oldbalance + Decimal('5.0')
+
+        recipients  = [ {"address": receiveraddr, "amount": Decimal('5.0')},
+                        {"address": receiveraddr2, "amount": Decimal('5.0')}  ]
+
+        myopid = self.nodes[0].z_sendmanywithchangetosender(senderaddr, recipients, minconf, fee)
+        mytxid = wait_and_assert_operationid_status(self.nodes[0], myopid)
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_all()
+        calculatedSenderBalance = calculatedSenderBalance - fee - receiverExpectedBalance * 2
```

Comment:
should be here correct is next?
        calculatedSenderBalance = calculatedSenderBalance - fee - 5.0 * 2
as 'receiverExpectedBalance' includes 'receiveraddr_oldbalance' that can be not 0

Reply:
thanks. fixed.